ui/apps/platform/src/Containers/Policies/Wizard/Step1/PolicyCategoriesSelectField.tsx

```diff
@@ -1,35 +1,70 @@
-import React, { useState, useEffect, ReactElement } from 'react';
-import { FormGroup, FormHelperText, HelperText, HelperTextItem } from '@patternfly/react-core';
-import { Select, SelectOption } from '@patternfly/react-core/deprecated';
+import React, { useState, useEffect, useMemo, ReactElement } from 'react';
+import {
+    FormGroup,
+    FormHelperText,
+    HelperText,
+    HelperTextItem,
+    Select,
+    SelectOption,
+    SelectList,
+    MenuToggle,
+    MenuToggleElement,
+    TextInputGroup,
+    TextInputGroupMain,
+    TextInputGroupUtilities,
+    ChipGroup,
+    Chip,
+    Button,
+} from '@patternfly/react-core';
 import { useField } from 'formik';
 
 import { getPolicyCategories } from 'services/PolicyCategoriesService';
 import { PolicyCategory } from 'types/policy.proto';
+import { TimesIcon } from '@patternfly/react-icons';
 
 function PolicyCategoriesSelectField(): ReactElement {
     const [policyCategories, setPolicyCategories] = useState<PolicyCategory[]>([]);
-    // manage state for Categories select below
-    const [isCategoriesOpen, setIsCategoriesOpen] = useState(false);
-
+    const [isOpen, setIsOpen] = useState(false);
+    const [inputValue, setInputValue] = useState('');
     const [field, , helpers] = useField('categories');
+    // Used to temporarily prevent dropdown from closing after selecting an item to maintain multi-select UX
+    const [preventClose, setPreventClose] = useState(false);
+
+    // Memoize to prevent unnecessary re-renders of dependent components when field.value reference changes
+    const selectedCategories: string[] = useMemo(
+        () => (field.value as string[]) || [],
+        [field.value]
+    );
 
-    function onCategoriesToggle(isOpen) {
-        setIsCategoriesOpen(isOpen);
-    }
+    const onToggle = () => {
+        setIsOpen(!isOpen);
+    };
 
-    function onSelectHandler(selectedCategories) {
-        return (event, selection) => {
-            const newSelectedCategories = selectedCategories.includes(selection)
-                ? selectedCategories.filter((item) => item !== selection)
-                : [...selectedCategories, selection];
-            helpers.setValue(newSelectedCategories);
-        };
-    }
+    const onSelect = (
+        _event: React.MouseEvent<Element, MouseEvent> | undefined,
+        value: string | number | undefined
+    ) => {
+        if (typeof value === 'string' && !selectedCategories.includes(value)) {
+            helpers.setValue([...selectedCategories, value]);
+            setPreventClose(true);
+            // Reset the preventClose flag after a brief delay to allow selection to complete
+            setTimeout(() => setPreventClose(false), 100);
```

Comment:
Having to use a `preventClose` feels pretty awkward, but it doesn't look like there is a clean way to do this given the new PatternFly components... 🤔 

Would it be possible to do something like
```ts
if (preventClose) {
    setPreventClose(false);
}
```
in the `onOpenChange` callback? If that worked identically, it would seem a little more deterministic than using a timeout here.